sdk/python/lib/pulumi/asset.py

```diff
@@ -81,41 +82,44 @@ class Archive(object):
 
 @known_types.asset_archive
 class AssetArchive(Archive):
+    assets: Dict[Any, Union[Asset, Archive]]
+
     """
     An AssetArchive is an archive created from an in-memory collection of named assets or other archives.
     """
-    def __init__(self, assets):
+    def __init__(self, assets: Dict[Any, Union[Asset, Archive]]) -> None:
         if not isinstance(assets, dict):
             raise TypeError("AssetArchive assets must be a dictionary")
-        for k, v in assets.items():
+        for _, v in assets.items():
```

Comment:
If the answer is yes, I suppose we should also check that `k` is a `basestring`?